101-machine-learning-create/azuredeploy.json

```diff
@@ -1,132 +1,114 @@
-        "australiaeast",
-        "brazilsouth",
-        "canadacentral",
-        "centralus",
-        "eastasia",
-        "eastus",
-        "eastus2",
-        "francecentral",
-        "japaneast",
-        "koreacentral",
-        "northcentralus",
-        "northeurope",
-        "southeastasia",
-        "southcentralus",
-        "uksouth",
-        "westcentralus",
-        "westus",
-        "westus2",
-        "westeurope"
-      ],
-      "metadata": {
-        "description": "Specifies the location for all resources."
-      }
-    }
-  },
-  "variables": {
-    "storageAccountName": "[concat('sa',uniqueString(resourceGroup().id))]",
-    "storageAccountType": "Standard_LRS",
-    "keyVaultName": "[concat('kv',uniqueString(resourceGroup().id))]",
-    "tenantId": "[subscription().tenantId]",
-    "applicationInsightsName": "[concat('ai',uniqueString(resourceGroup().id))]",
-    "containerRegistryName": "[concat('cr',uniqueString(resourceGroup().id))]"
-  },
-  "resources": [
-    {
-      "type": "Microsoft.Storage/storageAccounts",
-      "apiVersion": "2019-04-01",
-      "name": "[variables('storageAccountName')]",
-      "location": "[parameters('location')]",
-      "sku": {
-        "name": "[variables('storageAccountType')]"
-      },
-      "kind": "StorageV2",
-      "properties": {
-        "encryption": {
-          "services": {
-            "blob": {
-              "enabled": true
+    "variables": {
+        "storageAccountName": "[concat('sa',uniqueString(resourceGroup().id))]",
+        "storageAccountType": "Standard_LRS",
+        "keyVaultName": "[concat('kv',uniqueString(resourceGroup().id))]",
+        "tenantId": "[subscription().tenantId]",
+        "applicationInsightsName": "[concat('ai',uniqueString(resourceGroup().id))]",
+        "containerRegistryName": "[concat('cr',uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2019-04-01",
+            "name": "[variables('storageAccountName')]",
+            "location": "[if(or(equals(parameters('location'),'australiaeast'),equals(parameters('location'),'brazilsouth'),equals(parameters('location'),'canadacentral'),equals(parameters('location'),'centralus'),equals(parameters('location'),'eastasia'),equals(parameters('location'),'eastus'),equals(parameters('location'),'eastus2'),equals(parameters('location'),'francecentral'),equals(parameters('location'),'japaneast'),equals(parameters('location'),'koreacentral'),equals(parameters('location'),'northcentralus'),equals(parameters('location'),'northeurope'),equals(parameters('location'),'southeastasia'),equals(parameters('location'),'southcentralus'),equals(parameters('location'),'uksouth'),equals(parameters('location'),'westcentralus'),equals(parameters('location'),'westus'),equals(parameters('location'),'westus2'),equals(parameters('location'),'westeurope')),parameters('location'),'southcentralus')]",
```

Comment:
Hello Brian... 

Mate, I know, it doesn't look nice. My intention was to treat the available locations for the resource without change our standard:

 "location": {
        "type": "string",
        "defaultValue": "[resourceGroup().location]",
        "metadata": {
                "description": "Location for all resources."
            }
        },

But seems that it's more elegant use the parameter instead of tread as condition.
What is your consideration about it?